File: src/pattern/abstractFactory/listFactory/java/ListPage.java
package pattern.abstractFactory.listFactory.java;

import pattern.abstractFactory.factory.java.Item;
import pattern.abstractFactory.factory.java.Page;

public class ListPage extends Page {
    public ListPage(String title, String author){
        super(title, author);
    }

    @Override
    public String makeHTML(){
        StringBuffer buffer = new StringBuffer();

        buffer.append("<html>\n");
        buffer.append("<head><title>"+this.title+"</title></head>\n");
        buffer.append("<body>\n");
        buffer.append("<p>" + this.author + "</p>");
        for (Item it :
                this.content) {
            buffer.append(it.makeHTML());
        }

        buffer.append("</body>\n");
        buffer.append("</html>");


        return buffer.toString();
    }

}

File: src/pattern/abstractFactory/listFactory/kotlin/ListPage.kt
package pattern.abstractFactory.listFactory.kotlin

import pattern.abstractFactory.factory.kotlin.Item
import pattern.abstractFactory.factory.kotlin.Page

class ListPage(title: String, author: String): Page(title, author) {
    override fun makeHTML(): String {
        val buffer = StringBuffer()
        buffer.append("<html>" +
                "<head><title>${this.title}</title></head>" +
                "<body>")

        for (item: Item in this.content){
            buffer.append(item.makeHTML())
        }

        buffer.append("</body></html>")

        return buffer.toString()

    }
}